python/pyspark/pandas/data_type_ops/boolean_ops.py

```diff
@@ -50,7 +50,7 @@ def pretty_name(self) -> str:
     def add(self, left, right) -> Union["Series", "Index"]:
         if isinstance(right, numbers.Number):
             left = left.spark.transform(lambda scol: scol.cast(as_spark_type(type(right))))
-            return column_op(Column.__add__)(left, right)
+            return left + right
         elif isinstance(right, IndexOpsMixin) and is_numeric_index_ops(right):
             left = transform_boolean_operand_to_numeric(left, right.spark.data_type)
             return column_op(Column.__add__)(left, right)
```

Comment:
We can use `left + right` here as well?

Reply:
You are right, I was thinking it caused an infinite recursion. `left + right`s are applied.